### PR TITLE
Fixes bug found during release of 22.9.0

### PR DESCRIPTION
**Pull request by travishathaway**
This is a proposed fix for an issue initially discovered while creating a new pull request for the [AnacondaRecipes/conda-feedstock](https://github.com/AnacondaRecipes/conda-feedstock) repository.

Pull request:

- https://github.com/AnacondaRecipes/conda-feedstock/pull/12

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda/core/portability.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
                  br'(.*)'  # the rest of the line can contain option flags
                  br')$')  # end whole_shebang group
 
-MAX_SHEBANG_LENGTH = 127 if on_linux else 512  # Not used on Windows
+MAX_SHEBANG_LENGTH = 127 if (on_linux or on_mac) else 512  # Not used on Windows
```
Comment (kenodegard) [inline conda/core/portability.py:31]: Flip it?

```suggestion
MAX_SHEBANG_LENGTH = 512 if on_win else 127
```

Comment (travishathaway) [reply conda/core/portability.py:31]: Yeah, that's easy to parse! (for humans)

File: conda/core/portability.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 from ..auxlib.ish import dals
 from ..base.constants import PREFIX_PLACEHOLDER
 from ..base.context import context
-from ..common.compat import on_win, on_linux
+from ..common.compat import on_win, on_linux, on_mac
```
Comment (kenodegard) [inline conda/core/portability.py:16]: Should be able to remove both of these now:

```suggestion
from ..common.compat import on_win
```

**Review by beeankha — DISMISSED** [review]
👍

**Review by kenodegard — CHANGES_REQUESTED** [review]
As a bugfix this needs a news snippet

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

File: conda/core/portability.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
                  br'(.*)'  # the rest of the line can contain option flags
                  br')$')  # end whole_shebang group
 
-MAX_SHEBANG_LENGTH = 127 if on_linux else 512  # Not used on Windows
+MAX_SHEBANG_LENGTH = 512 if on_win else 127
```
Comment (jaimergp) [inline conda/core/portability.py:31]: Windows has no concept of shebang, so if the intent here is to have 127 for both Linux and Mac, I'd say we keep it simple:

```suggestion
MAX_SHEBANG_LENGTH = 127
```

Comment (kenodegard) [reply conda/core/portability.py:31]: Ah, in which case I think we all misunderstood the purpose of this code then, this change shouldn't be merged.

cc @jezdez @travishathaway

Comment (travishathaway) [reply conda/core/portability.py:31]: @kenodegard, I think this still needs to be merged. My suggestion would be to go with what @jaimergp put forth. It fixes a bug that was found during a build test.

**Pull request closed without merging**